components/tidb_query_executors/src/index_scan_executor.rs

```diff
@@ -727,6 +763,18 @@ impl IndexScanExecutorImpl {
         Ok(())
     }
 
+    #[inline]
+    fn process_physical_table_id_column(
+        &mut self,
+        key: &[u8],
+        columns: &mut LazyBatchColumnVec,
+    ) -> Result<()> {
+        let table_id = table::decode_table_id(key)?;
+        let col_index = columns.columns_len() - 1;
+        columns[col_index].mut_decoded().push_int(Some(table_id));
```

Comment:
Why `physical_table_id_column` must be the last column?

Reply:
When added to the list of columns in TiDB, it will be last (idea was taken from the Global Index handling of / EXTRA_PARTITION_ID_COL_ID). Also I don't really understand why the order seems to be important for index scan, but not for table scan.

@andylokandy do you want me to try to change the code so it is not dependent of column order?

Also I have not really understood why TiKV should be aware of EXTRA_PARTITION_ID_COL_ID at all, since I wonder if it is not just a simple i64 column for TiKV, that TiDB should handle fully?

Reply:
> do you want me to try to change the code so it is not dependent of column order?

I'd like to make it an explicit const number denoting the nth column counting back from the last one.

> Also I have not really understood why TiKV should be aware of EXTRA_PARTITION_ID_COL_ID at all, since I wonder if it is not just a simple i64 column for TiKV, that TiDB should handle fully?

@wjhuang2016 Do you have any idea?

Reply:
> > do you want me to try to change the code so it is not dependent of column order?
> 
> I'd like to make it an explicit const number denoting the nth column counting back from the last one.
> 
> > Also I have not really understood why TiKV should be aware of EXTRA_PARTITION_ID_COL_ID at all, since I wonder if it is not just a simple i64 column for TiKV, that TiDB should handle fully?
> 
> @wjhuang2016 Do you have any idea?

No， @tiancaiamao PTAL

Reply:
An explicit order looks good to me, as this module is a tidb_query module and its logic can be highly coherent with the TiDB.